### PR TITLE
feat(ui): mark layer with bad projections

### LINKED ISSUE #626: feat(ui): display a notification when a layer cannot be displayed in the current basemap
As was found in #491, some (tile?) layers might not be shown in certain basemap projections. We need a way to indicate this in the legend.

### PR DESCRIPTION
**Pull request by AleksueiR**
## Description
Layers that cannot be displayed in the current projection are marked with a "bad-projection" flag and sad icon.
Also removes useless now layer blueprint user options class.

Closes #626

## Testing
Eyeballs.

## Documentation
Inline comments.

## Checklist

- [x] commits messages follow the guidelines
- [x] code passes unit tests
- [x] release notes have been updated
- [x] PR targets the correct release version

Remember, it is a *muffin offence* to open a PR with any of the above checklist items incomplete.

Please keep the original issue up to date with the final solution, expected behaviour, and any additional notes for testers

---
This change is [Reviewable](https://reviewable.io/reviews/fgpv-vpgf/fgpv-vpgf/1966)

### CODE REVIEW

**Comment by alyec** [comment]
:lgtm:

---

Reviewed 14 of 14 files at r1.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

*[src/app/geo/legend-block.class.js, line 152 at r1](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/1966#-KmWVDqwTGGbKS9UW3Uc:-KmWVDqxYdB5pyLdBubv:byn4vhx) ([raw file](https://github.com/fgpv-vpgf/fgpv-vpgf/blob/a8c9b5f0a3bd997d8f89960b559aab7d7ac51843/src/app/geo/legend-block.class.js#L152)):*
> ```JavaScript
>         get validProjection () { return this._validProjection; }
>         /**
>          * Checks if the spacial reference of the layer matches the spacial reference of the current basemap.
> ```

typo: spatial

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/1966#-:-KmWWr4pN4Sedbc64b_l:bnfp4nl)*

**Comment by AleksueiR** [comment]
Review status: 13 of 14 files reviewed at latest revision, 1 unresolved discussion.

---

*[src/app/geo/legend-block.class.js, line 152 at r1](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/1966#-KmWVDqwTGGbKS9UW3Uc:-KmWZ4GnVqaDccYza47N:b-896fix) ([raw file](https://github.com/fgpv-vpgf/fgpv-vpgf/blob/a8c9b5f0a3bd997d8f89960b559aab7d7ac51843/src/app/geo/legend-block.class.js#L152)):*
<details><summary><i>Previously, alyec (Aly Merchant) wrote…</i></summary><blockquote>

typo: spatial
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/1966)*

**Comment by alyec** [comment]
Reviewed 1 of 1 files at r2.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/1966)*

**Comment by james-rae** [comment]
Reviewed 13 of 14 files at r1, 1 of 1 files at r2.
Review status: all files reviewed at latest revision, all discussions resolved, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/1966)*

**Comment by james-rae** [comment]
Reviewed 5 of 5 files at r3.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/1966)*

**Comment by alyec** [comment]
Reviewed 5 of 5 files at r3.
Review status: :shipit: all files reviewed at latest revision, all discussions resolved, all commit checks successful.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/1966)*

File: src/app/geo/legend-block.class.js [hunk 1]
```diff
@@ -518,6 +542,10 @@ function LegendBlockFactory($q, Geo, layerRegistry, gapiService, configService,
                 'rv-error': () => {
                     _addReload();
                     return 'error';
+                },
+                'rv-bad-projection': () => {
+                    _removeReload()
```
Comment (milesap) [inline src/app/geo/legend-block.class.js]: Missing a `;`

Comment (AleksueiR) [inline src/app/geo/legend-block.class.js]: Very strange, how come eslint didn't catch that.

**Pull request merged**